tests/backup/backup_restore_basic_test.go

```diff
@@ -1442,3 +1442,220 @@ var _ = Describe("{BackupMultipleNsWithSameLabel}", func() {
+			clusterStatus, err = Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			clusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+		})
+
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			var sem = make(chan struct{}, numberOfSimultaneousBackups)
+			var wg sync.WaitGroup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD("Taking %d backups", numberOfBackups)
+			for backupLocationUID, backupLocationName := range backupLocationMap {
+				for _, namespace := range bkpNamespaces {
+					for i := 0; i < numberOfBackups; i++ {
+						sem <- struct{}{}
+						time.Sleep(timeBetweenConsecutiveBackups)
+						backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+						backupNames = append(backupNames, backupName)
+						wg.Add(1)
+						go func(backupName string) {
+							defer GinkgoRecover()
+							defer wg.Done()
+							defer func() { <-sem }()
+							err = CreateBackup(backupName, SourceClusterName, backupLocationName, backupLocationUID, []string{namespace},
+								labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+							dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup creation: %s", backupName))
+						}(backupName)
+					}
+					wg.Wait()
+				}
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
+
+		Step("Remove the backup locations where backups were taken", func() {
+			log.InfoD("Remove backup locations where backups were taken")
+			// Issue a remove backup location call
+			for backupLocationUID, customBackupLocationName = range backupLocationMap {
+				err := DeleteBackupLocation(customBackupLocationName, backupLocationUID, orgID, false)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Deleting backup location %s", customBackupLocationName))
+
+				// Wait until backup location is removed
+				backupLocationDeleteStatusCheck := func() (interface{}, bool, error) {
+					ctx, err := backup.GetAdminCtxFromSecret()
+					log.FailOnError(err, "Fetching px-central-admin ctx")
+					status, err := IsBackupLocationPresent(customBackupLocationName, ctx, orgID)
+					if err != nil {
+						return "", true, fmt.Errorf("backup location %s still present with error %v", customBackupLocationName, err)
+					}
+					if status == true {
+						return "", true, fmt.Errorf("backup location %s is not deleted yet", customBackupLocationName)
+					}
+					return "", false, nil
+				}
+				_, err = task.DoRetryWithTimeout(backupLocationDeleteStatusCheck, 3*time.Minute, 30*time.Second)
```

Comment:
@mkoppal-px Can you please provide some info here ^

Reply:
@tthurlapati-px `defer ginkgoRecover()` is needed only when we are running in a gorountine. But here it will be running the main thread. So not needed.

Reply:
@mkoppal-px but in the code for `task.DoRetryWithTimeout`, I see that the callback is being invoked in a goroutine (https://vscode.dev/github/portworx/torpedo/blob/b4594a71680262773211f1a098da604d09812545/vendor/github.com/portworx/sched-ops/task/task.go#L39)

Reply:
- After a discussion with Mithun, we confirmed that this is definitely a previously missed error. But we decided that this error needs to be fixed for many other functions, so we'll fix it in a separate PR.

- This code can be made faster as there's no need for getting `ctx` each time. **Following is faster**:
```go
                // Wait until backup location is removed
				ctx, err := backup.GetAdminCtxFromSecret()
				log.FailOnError(err, "Fetching px-central-admin ctx")
				backupLocationDeleteStatusCheck := func() (interface{}, bool, error) {
					status, err := IsBackupLocationPresent(customBackupLocationName, ctx, orgID)
					if err != nil {
						return "", true, fmt.Errorf("backup location %s still present with error %v", customBackupLocationName, err)
					}
					if status == true {
						return "", true, fmt.Errorf("backup location %s is not deleted yet", customBackupLocationName)
					}
					return "", false, nil
				}
				_, err = task.DoRetryWithTimeout(backupLocationDeleteStatusCheck, 3*time.Minute, 30*time.Second)
```

Reply:
okay done.